stock_dynamic_routing/tests/test_routing_pull.py

```diff
@@ -910,3 +916,138 @@ def test_change_dest_move_source_chain(self):
+                "picking_type_id": self.wh.pick_type_id.id,
+                "rule_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "method": "pull",
+                            "picking_type_id": priority_pick_type.id,
+                            "rule_domain": [("priority", "=", "3")],
+                        },
+                    )
+                ],
+            }
+        )
+
+        pick_picking, customer_picking = self._create_pick_ship(
+            self.wh, [(self.product1, 30)]
+        )
+        pick_picking.move_lines.priority = "3"
+
+        self._update_product_qty_in_location(self.location_hb_1_1, self.product1, 5)
+        self._update_product_qty_in_location(self.location_hb_1_2, self.product1, 7)
+        self._update_product_qty_in_location(self.location_shelf_1, self.product1, 10)
+        self._update_product_qty_in_location(self.location_shelf_2, self.product1, 8)
+
+        pick_picking.action_assign()
+
+        # all the moves are moved to the priority picking type -> original pick
+        # cancelled
+        self.assertEqual(pick_picking.state, "cancel")
+
+        customer_moves = customer_picking.move_lines
+        pick_picking = customer_moves.move_orig_ids.picking_id
+        pick_moves = pick_picking.move_lines
+
+        shelf_move = pick_moves.filtered(lambda m: m.product_uom_qty == 18)
+        handover_move = pick_moves.filtered(lambda m: m.product_uom_qty == 12)
+
+        # At this point, we should have 3 stock.picking:
+        #
+        # +-------------------------------------------------------------------+
+        # | HO/xxxx  Assigned                                                 |
+        # | Stock → Stock/Handover                                            |
+        # | 12x Product Highbay  → Stock/Handover (available)                 |
+        # |   lines: 5 from Bay1/Bin1, 7 from Bay1/Bin2                       |
+        # +-------------------------------------------------------------------+
+        #
+        # +-------------------------------------------------------------------+
+        # | PRI/xxxx Waiting                                                  |
+        # | Stock → Output                                                    |
+        # | 12x Product Stock → Output (waiting on the Handover moves)        |
+        # | 18x Product Stock → Output (available)                            |
+        # |   lines: 10 from Stock/Shelf1, 8 from Stock/Shelf2                |
+        # +-------------------------------------------------------------------+
+        #
+        # +-------------------------------------------------+
+        # | OUT/xxxx Waiting                                |
+        # | Output → Customer                               |
+        # | 18x Product Output → Customer (waiting)         |
+        # | 12x Product Output → Customer (waiting)         |
```

Comment:
Question: is it possible to get only one move of 30 units waiting for the others?

Reply:
I don't remember why the next move is split, this is what I observed, so I documented it this way.

Reply:
Probably linked to this PR https://github.com/OCA/wms/pull/232